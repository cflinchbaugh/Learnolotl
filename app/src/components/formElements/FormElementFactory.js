import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';

import InputField from 'formElements/InputField';

const StyleWrapper = styled.div`
`

class FormElementFactory extends Component {
    constructor(props) {
        super(props);

        this.handleFormElementChange = this.handleFormElementChange.bind(this);
    }

    render() {
        let builtFormElements = this._buildFormElements();
        
        return (
            <StyleWrapper>
                {builtFormElements}
            </StyleWrapper>

        )
    }

    handleFormElementChange(e) {
        let formElement = e.currentTarget,
            formElementData = {
                [formElement.id]: formElement.value
            }

        this.props.handleInputChange(formElementData);
    }

    

    _buildFormElements() {
        return this.props.formElements.map((element, idx) => {
            if (element.type === 'input') {
                let id = idx,
                    processedValue = this._processValue(id),
                    inputFieldData = {
                        key: idx,
                        id: id,
                        onChange: this.handleFormElementChange,
                        autoComplete: 'off',
                        value: processedValue
                    }

                //inputFieldData must come after ...element.data to override the passed in id with the id+idx
                return <InputField {...element.data} {...inputFieldData}/>
            }
        });
    }

    _processValue(id) {
        let processedValue = this.props.formElementData !== null ? this.props.formElementData[id] : '';

        if (typeof(processedValue) === 'undefined') {
            processedValue = '';
        }

        return processedValue;
    }
}

const mapStateToProps = state => ({
});

export default connect(mapStateToProps, { })(FormElementFactory);