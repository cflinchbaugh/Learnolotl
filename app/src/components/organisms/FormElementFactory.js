import React, { Component } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';

import InputField from 'atoms/formElements/InputField';

const StyleWrapper = styled.div`
`

class FormElementFactory extends Component {
    render() {
        let builtFormElements = this._buildFormElements();
        
        return (
            <StyleWrapper>
                {builtFormElements}
            </StyleWrapper>

        )
    }

    _buildFormElements() {
        console.log("Props: " + this.props);
        return <InputField />
    }
}

const mapStateToProps = state => ({
});

export default connect(mapStateToProps, { })(FormElementFactory);