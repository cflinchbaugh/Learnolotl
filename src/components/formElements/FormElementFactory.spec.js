import React from 'react';
import {shallow} from 'enzyme';
import {FormElementFactory} from './FormElementFactory';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('FormElementFactory', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const formElementFactoryData = {
            formElements: [
                {
                    type: ''
                }
            ]
        }

        const tree = renderer.render(
            <FormElementFactory {...formElementFactoryData}/>
        );

        expect(tree).toMatchSnapshot();
    });

});