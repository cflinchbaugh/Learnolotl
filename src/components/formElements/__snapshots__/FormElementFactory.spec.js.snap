// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormElementFactory renders and matches snapshot 1`] = `
<styled.div>
  
</styled.div>
`;
